### PR TITLE
Remove rate limiting from app service senders and fix get_or_create_user requester

### PR DESCRIPTION
**Pull request by Rugvip**
Removing the rate limiting from app services may be a problem for existing ASes that are relying on the rate limiting in synapse, although I doubt that's an issue since this is only for the AS sender.
We could do some config, but it's probably not worth the hassle of having an extra config parameter.

### CODE REVIEW

**Comment by erikjohnston** [comment]
ok to test

**Comment by erikjohnston** [comment]
@matrixbot retest this please

File: synapse/handlers/_base.py [hunk 1]
```diff
@@ -55,8 +55,14 @@ def __init__(self, hs):
 
     def ratelimit(self, requester):
         time_now = self.clock.time()
+        user_id = requester.user.to_string()
+
+        defer = self.store.get_app_service_by_user_id(user_id)
+        if defer.result is not None:
+            return  # do not ratelimit app service senders
```
Comment (Rugvip) [inline synapse/handlers/_base.py]: Each that's probably a better option, I'll have a look

**Comment by Rugvip** [comment]
didn't even have to change the docs x)

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Comment by Rugvip** [comment]
What is Jenkins running other than `python setup.py test`? Not managing to get the same failures locally.

**Comment by erikjohnston** [comment]
The unit tests are run via `trial tests`, the sytests are run via https://github.com/matrix-org/sytest

**Pull request merged**